.github/workflows/poetry-autoupdate.yml

```
@@ -0,0 +1,39 @@
+---
+name: Check for poetry updates and open PR
+on:
+  workflow_dispatch: ~
+  schedule:
+    - cron: '0 6 * * *'
+
+jobs:
+  update-dep:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Update poetry to latest version.
+        run: |
+          VERSION="$(curl -f -s -H "Accept: application/json" "https://api.github.com/repos/python-poetry/poetry/releases/latest" | sed -n '/tag_name/p' | cut -d: -f2 | sed 's/[",[:space:]]//g')"
+
+          printf 'POETRY_VERSION=%s\n' "$VERSION" >> $GITHUB_ENV
+
+          if [ -n "$VERSION" ] ; then
+            sed -i "/^POETRY_VERSION=/s/=.*$/=\"$VERSION\"/" scripts/build.sh
+          fi
+
+      - name: Open PR
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message:
+            '[bot] Updated poetry for version ${{ env.POETRY_VERSION }}.'
+          title: '[bot] Updated poetry for version ${{ env.POETRY_VERSION }}.'
```

I prefer using the imperative mood here
```suggestion
            '[bot] Update poetry to version ${{ env.POETRY_VERSION }}.'
          title: '[bot] Update poetry to version ${{ env.POETRY_VERSION }}.'
```